src/IonTextReader.ts

```diff
@@ -87,41 +89,62 @@ export class TextReader implements Reader {
     }
   }
 
-  next() {
-    this._raw = undefined;
-    if (this._raw_type === EOF) {
-      return undefined;
+    isIVM(input : string) : boolean {
+        if(this.depth() > 0) return false;
+        if(input.length < 6 || this.annotations().length > 0) return false;
+        let prefix = "$ion_";
+        for(let i = 0; i < prefix.length; i++){
+            if(prefix.charAt(i) !== input.charAt(i)) return false;
+        }
+        if(input !== "$ion_1_0") throw new Error("Only ion version 1.0 is supported.");
+        return true;
     }
 
-    let should_skip: boolean =
-      this._raw_type !== BEGINNING_OF_CONTAINER
-      && !this.isNull()
-      && this._type
-      && this._type.container;
-    if (should_skip) {
-      this.skip_past_container();
+    isLikeIVM() : boolean {
+      return false;
     }
 
+next() {
+    this._raw = undefined;
+    if (this._raw_type === EOF) return undefined;
+
+    let should_skip: boolean =
+    this._raw_type !== BEGINNING_OF_CONTAINER
+    && !this.isNull()
+    && this._type
+    && this._type.container;
+    if (should_skip) this.skip_past_container();
+
     let p: ParserTextRaw = this._parser;
     for (;;) {
-      this._raw_type = p.next();
-      if (this._depth > 0) break;
-      if (this._raw_type === T_IDENTIFIER) {
-        this.load_raw();
-        if (this._raw !== IVM.text) break;
-        this._symtab = defaultLocalSymbolTable();
-      }
-      else if (this._raw_type === T_STRUCT) {
-        if (p.annotations().length !== 1) break;
-        if (p.annotations()[0] != ion_symbol_table) break;
-        this._symtab = makeSymbolTable(this._cat, this);
-      }
-      else {
-        break;
-      }
+        this._raw_type = p.next();
+        if (this._raw_type === T_IDENTIFIER) {
+            if (this._depth > 0) break;
+            this.load_raw();
+            //if(this._raw !== "$ion_1_0") break;
+            if (!this.isIVM(this._raw)) break;
+            this._symtab = defaultLocalSymbolTable();
+            this._raw = undefined;
+            this._raw_type = undefined;
+        }else if (this._raw_type === T_STRING1) {
+            if (this._depth > 0) break;
+            this.load_raw();
+            if (this._raw !== "$ion_1_0") break;
```

Comment:
Is this needed?

Reply:
yes

Reply:
Oh because T_STRING1 mean's single-quoted symbol. Got it.